components/script/dom/globalscope.rs

```diff
@@ -3310,6 +3312,28 @@ impl GlobalScope {
         }
         unreachable!();
     }
+
+    pub(crate) fn report_csp_violations(&self, violations: Vec<Violation>) {
+        for violation in violations {
+            let sample = match violation.resource {
+                ViolationResource::Inline { .. } | ViolationResource::Url(_) => None,
+                ViolationResource::TrustedTypePolicy { sample } => Some(sample),
+            };
+            let report = CSPViolationReportBuilder::default()
+                .resource("eval".to_owned())
+                .sample(sample)
+                .effective_directive(violation.directive.name)
+                .report_only(false)
+                .source_file("".to_string())
+                .line_number(0)
+                .column_number(0)
```

Comment:
These are the default values, so we don't need to explicitly set them with the builder pattern.